.github/workflows/ci.yaml

```diff
@@ -0,0 +1,36 @@
+name: ci
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+
+jobs:
+  markdown-lint:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: xt0rted/markdownlint-problem-matcher@v2
+      - uses: DavidAnson/markdownlint-cli2-action@v9
+        with:
+          globs: "**/*.md"
+
+  notify:
+    runs-on: ubuntu-latest
+    if: ${{always()}} && ${{failure()}} && ${{ github.event_name }} == 'push'
```

Comment:
I haven't tested this change out, but I noticed this isn't actually working in the sgx repo.
The sgx repo has
```yaml
if: failure() && ${{ github.event_name }} == 'push'
```
it maybe that the `always` part is unecessary and the if condition is enough to allow the needs to follow through when previous steps fail.  I also think the github context variables must have changed `github.repository_url` doesn't seem to exist in docs, neither does `github.pusher` and `github.url`